### DISCUSSION TITLE
Error '-1' when using HTTPClient with existing *tlsClient

### DISCUSSION THREAD

**Discussion by ksaye** [arendst/Tasmota#12073] [Q&A] [answered]
I have contributed code to [xdrv_02_mqtt.ino](https://github.com/arendst/Tasmota/blob/development/tasmota/xdrv_02_mqtt.ino) that uses the HTTPClient to make REST calls over SSL.

If I create the ```HTTPClient httpsClient```, then ```httpsClient.begin(*tlsClient, dPSURL);``` to the existing tlsClient before the ```MqttClient.setClient(*tlsClient);```, it works fine.  

If I try ```httpsClient.begin(*tlsClient, dPSURL);``` after the ```MqttClient.setClient(*tlsClient);```, I get the response code -1.

To overcome this, I have created my own ```BearSSL::WiFiClientSecure_light *tlsHttpsClient = new BearSSL::WiFiClientSecure_light(1024,1024);```, which always works but at the cost of about 5K of Heap.

Any thoughts or experience as to why this issues exist?

**Comment by s-hadinger** [comment, marked as answer]
Each TLS connection needs to keep data about the connection. You can't use the same for 2 different connections.

That's why we don't support HTTPS in Tasmota, the cost is too high in heap memory. ESP32 is more favorable.

**Reply by ksaye** [reply]
Thanks and that is what I experienced.
